src/dotnet/APIView/APIViewWeb/Managers/OpenSourceRequestManager.cs

```diff
@@ -0,0 +1,127 @@
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace APIViewWeb.Managers
+{
+    public class OpenSourceRequestManager : IOpenSourceRequestManager
+    {
+        static readonly string _ossApiUrl = "https://repos.opensource.microsoft.com/api/people/links/github/{GithubUser}?api-version=2019-10-01";
+        static readonly HttpClient _ossClient = new();
+        static readonly HttpClient _tokenClient = new();
+        static TelemetryClient _telemetryClient = new(TelemetryConfiguration.CreateDefault());
+
+        private readonly string _aadClientId;
+        private readonly string _aadClientSecret;
+        private readonly string _aadTenantId;
+        private readonly string _tokenApiUrl;
+
+        private DateTime _tokenExpiry;
+        private string _token;
+    
+        public OpenSourceRequestManager(IConfiguration configuration)
+        {
+            _aadClientId = configuration["opensource-aad-app-id"] ?? "";
+            _aadClientSecret = configuration["opensource-aad-client-secret"] ?? "";
+            _aadTenantId = configuration["opensource-aad-tenant-id"] ?? "";
+            _tokenApiUrl = $"https://login.microsoftonline.com/{_aadTenantId}/oauth2/token";
+        }
+
+        public async Task<OpenSourceUserInfo> GetUserInfo(string githubUserId)
+        {
+            await SetHeaders();
+            int counter = 0;
+            bool retryStatus = true;
+
+            while(retryStatus && counter <= 5)
+            {
+                var resp = await _ossClient.GetAsync(_ossApiUrl.Replace("{GithubUser}", githubUserId));
+                if(resp.StatusCode == HttpStatusCode.TooManyRequests)
+                {
+                    var retryAfter = resp.Headers.RetryAfter?.ToString() ?? "60";
+                    _telemetryClient.TrackTrace($"Request to get user info from OSS manger has been throttled. Retry After: {retryAfter}, Retry count: {counter}");
+                    await Task.Delay(int.Parse(retryAfter) * 1000);
+                    counter++;
+                }
+                else if (resp.StatusCode == HttpStatusCode.OK)
+                {
+                    var content = await resp.Content.ReadAsStringAsync();
+                    return JsonConvert.DeserializeObject<OpenSourceUserInfo>(content);
+                }
+                else
+                {
+                    retryStatus = false;
+                    _telemetryClient.TrackTrace("Failed to get user info from Microsoft Open source management");
+                }
+            }
+            
+            return null;            
+        }
+
+        public async Task<bool> IsAuthorizedUser(string githubUserId)
+        {
+            var resp = await GetUserInfo(githubUserId);
+            if (resp == null)
+                return false;
+            // For now we only need to check if user info is availableon MS OSS
```

Comment:
```suggestion
            // For now we only need to check if user is part of MS open source data
```

Reply:
At some point we might want to consider also checking that they are in the microsoft or Azure org.